src/main/java/org/apache/sling/scripting/core/impl/jsr223/SlingScriptEngineManager.java

```diff
@@ -293,13 +293,19 @@ private ScriptEngineManager getInternalScriptEngineManager() {
 
     private void registerAssociations(ScriptEngineFactory factory) {
         for (String extension : factory.getExtensions()) {
-            internalManager.registerEngineExtension(extension, factory);
+            if (extension != null && !extension.isEmpty()) {
+                internalManager.registerEngineExtension(extension, factory);
+            }
         }
```

Comment:
There should be an else block for all three registration types which emits at least a Warn on the log in case null is given

Reply:
Updated. Thanks for the review.